tools/lsp/preview/drop_location.rs

```diff
@@ -162,13 +162,17 @@ fn calculate_drop_information_for_layout(
                         LogicalSize::new(new_midpoint - last_midpoint, geometry.size.height),
                     );
                     if hit_rect.contains(position) {
-                        let start = (c.origin.x - last_endpoint) / 2.0;
-                        let start_pos = last_endpoint
-                            + if start.floor() < geometry.origin.x {
-                                geometry.origin.x
-                            } else {
-                                start
-                            };
+                        let start_pos = if pos > 0 {
+                            let start = (c.origin.x - last_endpoint) / 2.0;
+                            last_endpoint
+                                + if start.floor() < geometry.origin.x {
+                                    geometry.origin.x
+                                } else {
+                                    start
+                                }
```

Comment:
This i guess could be simplified as

```suggestion
                            last_endpoint + if start.floor().max(geometry.origin.x)
```

Which is much more readable. Although it's a different behavior as it takes start.floor() instead of start, but i don't know if that was on purpose.

(and same in the other one)

Reply:
you need to use f32::max   https://github.com/slint-ui/slint/pull/5115#discussion_r1575120222